### ISSUE TITLE
[bug] Patching files on Windows behaves different then on Linux

### ISSUE THREAD

**Issue by madebr** [conan-io/conan#6676]
Applying a patch to a read-only file throws a `PermissionError` on Windows.
This does not happen on Linux.

applying (git) patches to read-only files should not throw errors.

### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10 versus Linux
  * Conan version: 1.22.3
  * Python version: 3.8.2

### Steps to reproduce (Include if Applicable)
The following python script shows the problem
```python
#!/usr/bin/env python

from conans import tools
import os

INPUT = "some_file"
PATCH = "patch.patch"

try:
    os.chmod(INPUT, 0o666)
    os.unlink(INPUT)
except FileNotFoundError:
    pass
try:
    os.unlink(PATCH)
except FileNotFoundError:
    pass

tools.save(INPUT, """\
line 1
line 2
line 3
line 4
line 6
line 7
line 8
line 9
line 10
""")

tools.save(PATCH, """\
--- some_file
+++ some_file
@@ -2,6 +2,7 @@
 line 2
 line 3
 line 4
+line 5
 line 6
 line 7
 line 8
""")

os.chmod(INPUT, 0o444)

tools.patch(patch_file=PATCH)

assert "\nline 5" in tools.load(INPUT)
```

### Logs (Executed commands with output) (Include/Attach if Applicable)

```
Traceback (most recent call last):
  File "test.py", line 46, in <module>
    tools.patch(patch_file=PATCH)
  File "C:\Users\maarten\AppData\Roaming\Python\Python38\site-packages\conans\client\tools\files.py", line 213, in patch
    if not patchset.apply(root=base_path, strip=strip, fuzz=fuzz):
  File "C:\Python38\lib\site-packages\patch_ng.py", line 1109, in apply
    os.unlink(backupname)
PermissionError: [WinError 5] Access is denied: b'some_file.orig'
```

**Comment by czoido** [comment]
Hi @madebr,
Thanks for reporting, I can confirm the issue and the behavior should be the same for all the operating systems. For example `git apply` will apply the patch for read-only files in both Linux and Windows changing the permissions of the files. cc @uilianries

**Comment by uilianries** [comment]
@madebr what do you expect instead?

**Comment by madebr** [comment]
> 
> 
> @madebr what do you expect instead?

*It just work:*

- patch applied
- permission bits same as before
- same behavior as the patch tool (of git)

**Comment by uilianries** [comment]
The PR https://github.com/conan-io/python-patch-ng/pull/13 should fixes it, but we will need a new release.

**Comment by madebr** [comment]
Great!
Will you also upstream it to https://github.com/techtonik/python-patch/?

**Comment by jgsogo** [comment]
Upstream looks like dead. We've forked the repo and we are no longer sending changes upstream, but if the project becomes alive again we may consider it.

**Comment by jgsogo** [comment]
A PR to `patch-ng` has been merged fixing this issue, it will be released (if nothing else is added) as `1.17.4`, but first we need to change (open a little bit?) the dependency in Conan source code for next release.

**Comment by jgsogo** [comment]
Closed via https://github.com/conan-io/conan/pull/6698, it will be released in 1.24
